### PR TITLE
Fix "collective" Payment Method currency issues

### PR DESCRIPTION
**Pull request by znarf**
Related to: https://github.com/opencollective/opencollective/issues/1865

### CODE REVIEW

File: server/models/Collective.js [hunk 1]
```diff
@@ -1925,7 +1925,7 @@ export default function (Sequelize, DataTypes) {
     const promises = [models.Member.create(member), this.update(updatedValues)];
 
     // Invalidate current collective payment method if there's one
-    const collectivePaymentMethod = await models.PaymentMethod.findOne({
+    await models.PaymentMethod.destroy({
```
Comment (znarf) [inline server/models/Collective.js:1928]: It's cleaner to just destroy everything.

File: migrations/20210201160014-fix-opencollective-paymentMethod-currency.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    // Fix Payment Methods currency not matching the Host currency
```
Comment (znarf) [inline migrations/20210201160014-fix-opencollective-paymentMethod-currency.js:5]: We're migrating but the currency should have no impact on the transaction in the future.

File: migrations/20210201160014-fix-opencollective-paymentMethod-currency.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      WHERE c."id" = pm."CollectiveId"
+      AND pm."service" = 'opencollective'
+      AND pm."type" = 'collective'
+      AND pm."deletedAt" IS NULL
+      AND c."deletedAt" IS NULL
+      AND h."id" = c."HostCollectiveId"
+      AND pm."currency" !=  h."currency"
+    `);
+    // Fix some transactions impacted by this
+    // easy cases where t."currency" = t."hostCurrency"
```
Comment (znarf) [inline migrations/20210201160014-fix-opencollective-paymentMethod-currency.js:19]: Intent is to convert transactions exposed by:
https://opencollective-metabase.herokuapp.com/question/190

File: migrations/20210201160014-fix-opencollective-paymentMethod-currency.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    await queryInterface.sequelize.query(`
+      UPDATE "Transactions" as t
+      SET "hostCurrencyFxRate" = 1, "amountInHostCurrency" = "amount", "netAmountInCollectiveCurrency" = "amount"
```
Comment (znarf) [inline migrations/20210201160014-fix-opencollective-paymentMethod-currency.js:22]: As we make sure we're not looking at transaction with fees, we can have this simple resolution.

**Review by Betree — APPROVED** [review]
Looking good!

**Pull request merged**